### DISCUSSION TITLE
Fine grain access to read only SBOM

### DISCUSSION THREAD

**Discussion by pritiprajapati314** [community/community#60486] [Apps, API and Webhooks] [answered]
### Select Topic Area

Question

### Body

Hey experts, I need your help. 
I am trying to find if there is way to get the list of dependencies (like package name and version), from a private repository in a github organization, currently I am exploring SBOM and manifest files. My requirement is to 
1. have the least amount to access possible just enough to get the SBOM or manifest file(like pom.xml, package.json) without being able to read the content of the repository. 
2. I am also reading it on a regular interval so I need this source to be as updated as possible.
3. Is there any other way rather than creating another repo, or creating a github action publishing this data in an external api.

Things I have explored : github action, webhooks, graphQL.
I know that github provides fine grain tokens and gives read only permission to content type access. I can fetch SBOM (but the only issue is that it also enables me to read the content of the repository). There is no other permission which has SBOM export access so this is my best option till now.

I would be  really thankful if someone could tell me any other service github provide which would be relevant to the tenet. Or could provide a confirmation that no other way is possible. 

Thank you in advance for your time.

**Comment by nirav-gajera** [comment, marked as answer]
1. Fine-grained access control: GitHub provides fine-grained access control through repository-level permissions and personal access tokens. You mentioned that you can fetch the SBOM but it allows you to read the content of the repository. Unfortunately, there is no specific permission that allows access to only the SBOM or manifest file without read access to the repository.
2. External API or service: One option could be to create a GitHub Action that retrieves the SBOM or manifest file from the repository and publishes the dependency information to an external API or service
3. Custom solution: If you have specific constraints or requirements, you might need to explore custom solutions. For example, you could create a separate repository that acts as a proxy or aggregator for dependency information. 
4. Third-party services: There are third-party services and tools available that provide dependency management and monitoring capabilities. These services often integrate with version control systems like GitHub and can analyze repositories to extract dependency information. They may offer features such as vulnerability scanning, license compliance checks, and dependency tracking.

**Reply by pritiprajapati314** [reply]
Thanks Nirva! Can you suggest some third-party services you mentioned above to start with, it would be really helpful. I have explored SNYK, but I think it only shows vulnerability and not the whole list, are there any other services that provide the complete list of dependencies.

**Reply by nirav-gajera** [reply]
1. WhiteSource: WhiteSource offers a comprehensive software composition analysis (SCA) solution that helps manage open source components and dependencies.
2. Dependabot: Dependabot, now owned by GitHub, is a service that automatically scans repositories for outdated dependencies and creates pull requests to update them. It supports multiple package managers and provides visibility into your project's dependencies.
3. FOSSA: FOSSA is a dependency management and license compliance tool that integrates with version control systems to analyze and track dependencies.
4. JFrog Artifactory: JFrog Artifactory is a universal artifact repository manager that supports package managers and provides dependency management capabilities.
5. Sonatype Nexus

**Reply by pritiprajapati314** [reply]
Thank you! really appreciate your help
